provider/servicefabric/servicefabric.go

```diff
@@ -0,0 +1,420 @@
+func (p *Provider) getServiceLabelsWithPrefix(service ServiceItemExtended, prefix string) map[string]string {
+	results := make(map[string]string)
+	for k, v := range service.Labels {
+		if strings.HasPrefix(k, prefix) {
+			results[k] = v
+		}
+	}
+	return results
+}
+
+func (p *Provider) isPrimary(instance sfsdk.ReplicaInstance) bool {
+	_, data := instance.GetReplicaData()
+	primaryString := "Primary"
+	if data.ReplicaRole == primaryString {
+		return true
+	}
+	return false
+}
+
+func (p *Provider) doesAppParamContain(app sfsdk.ApplicationItem, key, shouldContain string) bool {
+	value := p.getApplicationParameter(app, key)
+	if strings.Contains(value, shouldContain) {
+		return true
+	}
+	return false
+}
+
+func (p *Provider) getApplicationParameter(app sfsdk.ApplicationItem, key string) string {
+	for _, param := range app.Parameters {
+		if param.Key == key {
+			return param.Value
+		}
+	}
+	log.Errorf("Parameter %s doesn't exist in app %s", key, app.Name)
+	return ""
+}
+
+func (p *Provider) getDefaultEndpoint(instance sfsdk.ReplicaInstance) string {
+	id, data := instance.GetReplicaData()
+	endpoint, err := getDefaultEndpoint(data.Address)
+	if err != nil {
+		log.Warnf("No default endpoint for replica %s in service %s endpointData: %s", id, data.Address)
+		return ""
+	}
+	return endpoint
+}
+
+func (p *Provider) getNamedEndpoint(instance sfsdk.ReplicaInstance, endpointName string) string {
+	id, data := instance.GetReplicaData()
+	endpoint, err := getNamedEndpoint(data.Address, endpointName)
+	if err != nil {
+		log.Warnf("No names endpoint of %s for replica %s in endpointData: %s", endpointName, id, data.Address)
+		return ""
+	}
+	return endpoint
+}
+
+func isHealthy(instanceData sfsdk.ReplicaItemBase) bool {
+	if instanceData.ReplicaStatus == "Ready" || instanceData.HealthState != "Error" {
+		return true
+	}
+	return false
+}
+
+func hasHTTPEndpoint(instanceData sfsdk.ReplicaItemBase) bool {
+	_, err := getDefaultEndpoint(instanceData.Address)
+	return err == nil
+}
+func decodeEndpointData(endpointData string) (map[string]string, error) {
+	var endpointsMap map[string]map[string]string
+
+	if endpointData == "" {
+		return nil, errors.New("Endpoint data is empty")
+	}
+
+	err := json.Unmarshal([]byte(endpointData), &endpointsMap)
+	if err != nil {
+		return nil, err
+	}
+	endpoints, endpointsExist := endpointsMap["Endpoints"]
+	if !endpointsExist {
+		return nil, errors.New("Endpoint doesn't exist in endpoint data")
+	}
+
+	return endpoints, nil
+}
+
+func getDefaultEndpoint(endpointData string) (string, error) {
+	endpoints, err := decodeEndpointData(endpointData)
+	if err != nil {
+		return "", err
+	}
+	var defaultHTTPEndpointExists bool
+	var defaultHTTPEndpoint string
+	for _, v := range endpoints {
+		isHTTP := strings.Contains(v, "http")
+		if isHTTP {
+			defaultHTTPEndpoint = v
+			defaultHTTPEndpointExists = true
+			break
+		}
+	}
+	if !defaultHTTPEndpointExists {
+		return "", errors.New("No default endpoint found")
```

Comment:
Could you use a lower-case first letter?

https://github.com/golang/go/wiki/CodeReviewComments#error-strings